pkg/system/system.go

```diff
@@ -1116,7 +1116,19 @@ func Connect(isExternal bool) (*Client, error) {
 }
 
 // GetDesiredDBImage returns the desired DB image according to spec or env or default (in options)
-func GetDesiredDBImage(sys *nbv1.NooBaa) string {
+func GetDesiredDBImage(sys *nbv1.NooBaa, currentImage string) string {
+	// Postgres upgrade failure workaround
+	// if the current Postgres image is a postgresql-12 image, use NOOBAA_PSQL12_IMAGE. otherwise use GetdesiredDBImage
+	if IsPostgresql12Image(currentImage) {
+		psql12Image, ok := os.LookupEnv("NOOBAA_PSQL_12_IMAGE")
+		util.Logger().Warnf("The current Postgres image is a postgresql-12 image. using (%s)", psql12Image)
+		if !ok {
+			psql12Image = currentImage
```

Comment:
I am not sure if I understand the entire context. We can't return `options.Psql12Image` here?